node/network/bridge/src/lib.rs

```diff
@@ -392,37 +389,39 @@ async fn run_network<N: Network>(
 			Action::ReportPeer(peer, rep) => {
 				net.report_peer(peer, rep).await?;
 			}
-			Action::StartWork(relay_parent) => {
-				live_heads.push(relay_parent);
-				if let Some(view_update)
-					= update_view(&peers, &live_heads, &mut net, &mut local_view).await?
-				{
-					if let Err(e) = dispatch_update_to_all(
-						view_update,
-						event_producers.values(),
-						&mut ctx,
-					).await {
-						log::warn!("Aborting - Failure to dispatch messages to overseer");
-						return Err(e)
+			Action::ActiveLeaves(ActiveLeavesUpdate { activated, deactivated }) => {
+				for relay_parent in activated {
+					live_heads.push(relay_parent);
+					if let Some(view_update)
+						= update_view(&peers, &live_heads, &mut net, &mut local_view).await?
+					{
+						if let Err(e) = dispatch_update_to_all(
```

Comment:
One of the motivations of this change was to do only one `update_view`/`dispatch_update_to_all` when a new relay-parent comes in. That can be extracted to after the `live_heads` updates done by `activated`/`deactivated`.